### ISSUE TITLE
State group.present returns changed without changes

### ISSUE THREAD

**Issue by syphernl** [saltstack/salt#23198]
If a state contains a definition like:

```
vagrant:
  group.present:
    - addusers:
      - sample
```

it always shows up as changed, even though the user was already a member of the group:

```
----------
          ID: vagrant
    Function: group.present
      Result: True
     Comment: The following group attributes are set to be changed:
              addusers: ['sample']
     Started: 06:27:05.369906
    Duration: 11.113 ms
     Changes:   
              ----------
              Final:
                  All Changed applied successfully
```

Versions

```
                  Salt: 2014.7.5
                Python: 2.7.6 (default, Mar 22 2014, 22:59:56)
                Jinja2: 2.7.2
              M2Crypto: 0.21.1
        msgpack-python: 0.3.0
          msgpack-pure: Not Installed
              pycrypto: 2.6.1
               libnacl: Not Installed
                PyYAML: 3.10
                 ioflo: Not Installed
                 PyZMQ: 14.0.1
                  RAET: Not Installed
                   ZMQ: 4.0.4
                  Mako: 0.9.1
 Debian source package: 2014.7.5+ds-1ubuntu1
```

**Comment by jfindlay** [comment]
@syphernl, thanks for the report.

**Comment by yllar** [comment]
Any updates? Still broken in 2017.7.1

**Comment by KumarRajnish** [comment]
All,

Due to a misconfiguration on our Office365 tenant, all email addressed to your Ruckus Wireless email address (you@ruckuswireless.com) were delivered to a catch all mailbox.  This incident occurred between 11 PM to 8 AM (PST).  IT is delivering these emails with a “Reply-all” rule to all individuals on the email chain.  If you feel you are missing email attachments, please send email to helpdesk@ruckuswireless.com.

We apologize for any inconvenience this may have caused.

Regards,

IT

**Comment by twellspring** [comment]
Appears this is still a bug  2 years later.

**Comment by mchugh19** [comment]
I can't replicate on 2019.2
```
# cat group.sls
netdev:
  group.present:
    - addusers:
      - ubuntu
```

```
local:
----------
          ID: netdev
    Function: group.present
      Result: True
     Comment: Group netdev is present and up to date
     Started: 08:16:15.508052
    Duration: 1.989 ms
     Changes:

Summary for local
------------
Succeeded: 1
Failed:    0
------------
Total states run:     1
Total run time:   1.989 ms
```

**Comment by sagetherage** [comment]
closing as corrected in 2019.2
